### ISSUE TITLE
integrity check

### ISSUE THREAD

**Issue by mfeske** [nextcloud/user_external#119]
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- user_external
	- INVALID_HASH
		- appinfo/info.xml
		- lib/smb.php

Raw output
==========
Array
(
    [user_external] => Array
        (
            [INVALID_HASH] => Array
                (
                    [appinfo/info.xml] => Array
                        (
                            [expected] => 06b250c3a2c2ae01c8a2c4b963d6449c99645dca36ee197c3aa513fb7ac3e90e42dc1408ee6957408d85f9f9a973b8983719e9c15125d42211d43f0853c7bf75
                            [current] => e5da848ca44b3b60abb1c99a3d26455d78464f94ea1f5f522b07d642096a771dcde9a1a5289e271805292659edc42925a86b5d6ad22ffa98b79ecec5bfed69ff
                        )

                    [lib/smb.php] => Array
                        (
                            [expected] => 283a99667fd865e54423d575953423bc6e15862b7683885cb5675aafdb3b2cbc1400a1318ec81cdbd90ce827c3f839aa6daf6a0f6e3ac6bd2742947498dd14f9
                            [current] => 1c12005d92f88017564fb4135a532810aab33ffab22eac67c19501a59d213518e417cceb83f5e9ea4b448ea81fff3149da37eefb0100a61bce3140ce67f8decc
                        )

                )

        )

)

**Comment by violoncelloCH** [comment]
could you specify which version you're using?

**Comment by sweber83** [comment]
@violoncelloCH 
We are currently in the process of updating our nextcloud 15.0.12 -> 16.0.7 and are also having this issue with v0.8.0 of the user_external app.
Before updating we empty the apps folder, after the update we get this list of invalid files:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- user_external
	- INVALID_HASH
		- appinfo/info.xml
		- lib/smb.php

Raw output
==========
Array
(
    [user_external] => Array
        (
            [INVALID_HASH] => Array
                (
                    [appinfo/info.xml] => Array
                        (
                            [expected] => 06b250c3a2c2ae01c8a2c4b963d6449c99645dca36ee197c3aa513fb7ac3e90e42dc1408ee6957408d85f9f9a973b8983719e9c15125d42211d43f0853c7bf75
                            [current] => e5da848ca44b3b60abb1c99a3d26455d78464f94ea1f5f522b07d642096a771dcde9a1a5289e271805292659edc42925a86b5d6ad22ffa98b79ecec5bfed69ff
                        )

                    [lib/smb.php] => Array
                        (
                            [expected] => 283a99667fd865e54423d575953423bc6e15862b7683885cb5675aafdb3b2cbc1400a1318ec81cdbd90ce827c3f839aa6daf6a0f6e3ac6bd2742947498dd14f9
                            [current] => 1c12005d92f88017564fb4135a532810aab33ffab22eac67c19501a59d213518e417cceb83f5e9ea4b448ea81fff3149da37eefb0100a61bce3140ce67f8decc
                        )

                )

        )

)
```

**Comment by mfeske** [comment]
17.0.2

**Comment by gityeti** [comment]
Bonjour, same issue, upgrading NC 17.0.0 to 17.0.2
user_external 0.8.0
'# sudo -u www-data php occ integrity:check-app user_external
  - INVALID_HASH:
    - appinfo/info.xml:
      - expected: 06b250c3a2c...
      - current: e5da848ca44b...
    - lib/smb.php:
      - expected: 283a99667...
      - current: 1c12005d92...'

**Comment by sophie-h** [comment]
The two hashes in the packages `signature.json` are wrong. From the manually downloaded package:

```
user_external $ grep info.xml appinfo/signature.json
        "appinfo\/info.xml": "06b250c3a2c2ae01c8a2c4b963d6449c99645dca36ee197c3aa513fb7ac3e90e42dc1408ee6957408d85f9f9a973b8983719e9c15125d42211d43f0853c7bf75",
user_external $ sha512sum appinfo/info.xml 
e5da848ca44b3b60abb1c99a3d26455d78464f94ea1f5f522b07d642096a771dcde9a1a5289e271805292659edc42925a86b5d6ad22ffa98b79ecec5bfed69ff  appinfo/info.xml
```

Example for a correct hash

```
user_external $ grep app.php appinfo/signature.json
        "appinfo\/app.php": "13a9acc43e5b2774ff7d6f7de811baa9168584fbb10fec3a32c508b100e1d521e254aabdd9dd470ae395d3168a4335ba8da82550ffe2811eec957e1dc551510c",
user_external $ sha512sum appinfo/app.php 
13a9acc43e5b2774ff7d6f7de811baa9168584fbb10fec3a32c508b100e1d521e254aabdd9dd470ae395d3168a4335ba8da82550ffe2811eec957e1dc551510c  appinfo/app.php
```

**Comment by Wikinaut** [comment]
Same issue here. Please correct a.s.a.p., thanks.

**Comment by Wikinaut** [comment]
* 16 days old problem

Please fix it.

**Comment by mstinsky** [comment]
Same error after upgrading from 17.0.2 to 17.0.3.

**Comment by ChrisgammaDE** [comment]
I must report the same error after upgrading to 18.0.0

**Comment by apoger** [comment]
> I must report the same error after upgrading to 18.0.0

Same here!

**Comment by johansenmeister** [comment]
Same here. We are experiencing login issues too. New clients are timing out, existing mobile clients are timing out when the are uploading files.

**Comment by johansenmeister** [comment]
> Same here. We are experiencing login issues too. New clients are timing out, existing mobile clients are timing out when the are uploading files.

Are these log entries related?

[index] Error: Exception: Argument 1 passed to OC\Core\Controller\ClientFlowLoginV2Controller::grantPage() must be of the type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 170 at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/App.php line 125
   OC\AppFramework\Http\Dispatcher->dispatch(OC\Core\Controll ... {}, "grantPage")
1. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OC\\Core\\Contr ... r", "grantPage", OC\AppFramework\ ... {}, {_route: "core.C ... "})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "core.C ... "})
3. /var/www/nextcloud/lib/private/Route/Router.php line 299
   call_user_func(OC\AppFramework\ ... {}, {_route: "core.C ... "})
4. /var/www/nextcloud/lib/base.php line 1008
   OC\Route\Router->match("/login/v2/grant")
5. /var/www/nextcloud/index.php line 38
   OC::handleRequest()

GET /login/v2/grant
from 192.168.16.1 by Robin at 2020-02-02T18:43:57+01:00

**Comment by violoncelloCH** [comment]
yes, sorry, something went wrong with the signing process while packaging the release :upside_down_face: 
will be fixed with the next release :)

**Comment by violoncelloCH** [comment]
@johansenmeister that doesn't look related to user_external to me... anyway, please open separate issues for something that isn't clearly related :) I'd search the server repo for something related...

**Comment by violoncelloCH** [comment]
so I've uploaded a corrected release 0.8.0 which already fixes this for new installs/updates...
however as it has the same version number it won't update over the appstore if you already are on v0.8.0, because this is basically still the same version... if you want you can update manually (unpack the archive from https://apps.nextcloud.com/apps/user_external manually) or just copy over the new signature.json file in `user_external/appinfo/signature.json` (this should also be working)...
otherwise just wait for the next normal release :)

**Comment by andre-lx** [comment]
> so I've uploaded a corrected release 0.8.0 which already fixes this for new installs/updates...
> however as it has the same version number it won't update over the appstore if you already are on v0.8.0, because this is basically still the same version... if you want you can update manually (unpack the archive from https://apps.nextcloud.com/apps/user_external manually) or just copy over the new signature.json file in `user_external/appinfo/signature.json` (this should also be working)...
> otherwise just wait for the next normal release and ignore the failing integrity check until then :)

This works. 

I started with changing only the signature files hash, but didn't work. Replacing all the file content with the new one from the unpack works fine, and "All checks passed.". I didn't need to change all the files. Only the signature, as you mentioned.
```
user_external/appinfo/signature.json -> /var/www/nextcloud/apps/user_external/appinfo/signature.json
```

Thanks

**Comment by pm-pm** [comment]
I think if you remove and re-install the app via UI this should also work. (Of course you need to reconfigure.) ;-)
